ui/src/main/webapp/src/app/analysis-context/select-applications.component.ts

```diff
@@ -0,0 +1,51 @@
+import {Component, Input, EventEmitter, Output} from "@angular/core";
+import {RegisteredApplication} from "windup-services";
+
+@Component({
+    selector: 'wu-select-applications',
+    templateUrl: './select-applications.component.html'
+})
+export class SelectApplicationsComponent {
+    @Input()
+    availableApps:RegisteredApplication[];
+    _selectedApps:RegisteredApplication[];
+
+    @Output()
+    selectedAppsChange:EventEmitter<RegisteredApplication[]> = new EventEmitter<RegisteredApplication[]>();
+
+    @Input()
+    set selectedApps(apps:RegisteredApplication[]) {
+        this._selectedApps = apps;
+        this.selectedAppsChange.emit(this._selectedApps);
+    }
+
+    get selectedApps():RegisteredApplication[] {
+        return this._selectedApps;
+    }
+
+    get notAllSelected():boolean {
+        if (!this.availableApps)
+            return false;
+        if (!this._selectedApps)
+            return false;
+
+        return this._selectedApps.length != this.availableApps.length;
+    }
+
+    get someSelected():boolean {
+        return this._selectedApps ? this._selectedApps.length > 0 : false;
+    }
+
+    appsLabelCallback = (app: RegisteredApplication) => app.title;
+    equalsCallback = (a1: RegisteredApplication, a2: RegisteredApplication) => a1.id === a2.id;
+
+    selectAll() {
+        this.selectedApps = this.availableApps;
```

Comment:
You got me, I wanted to say you forgot to `emit`, but you use `set` property so it actually gets called... :)